test/Test/Toml/Gen.hs

```diff
@@ -0,0 +1,107 @@
+{-# LANGUAGE PatternSynonyms     #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications    #-}
+
+-- | This module contains all generators for @tomland@ testing.
+
+module Test.Toml.Gen
+       ( genVal
+       , genKey
+       , genPrefixMap
+       , genToml
+       ) where
+
+import Control.Monad (forM)
+
+import Hedgehog (MonadGen)
+
+import Toml.PrefixTree (pattern (:||), Key (..), Piece (..), PrefixMap, PrefixTree (..), fromList)
+import Toml.Type (AnyValue (..), TOML (..), Value (..))
+
+import qualified Data.HashMap.Strict as HashMap
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+
+----------------------------------------------------------------------------
+-- Common generators
+----------------------------------------------------------------------------
+
+type V = Int
+
+genVal :: MonadGen m => m V
+genVal = Gen.int (Range.constant 0 256)
+
+-- TODO: Arrays and Date.
+-- | Generates random value of 'AnyValue' type.
+genAnyValue :: MonadGen m => m AnyValue
+genAnyValue = do
+  randB <- Gen.bool
+  randI <- toInteger <$> Gen.int (Range.constantBounded @Int)
+  randF <- Gen.double $ Range.constant @Double (-1000000.0) 1000000.0
+  randS <- Gen.text (Range.constant 0 256) Gen.alphaNum
+  Gen.element
```

Comment:
I think it's better to use [`choice`](https://hackage.haskell.org/package/hedgehog-0.5.3/docs/Hedgehog-Gen.html#v:choice) combinator here. It can increase speed of tests. Because it looks to me that now this generator will generate every value when we need only one.